infra/build/functions/oss_fuzz_on_demand.py

```diff
@@ -0,0 +1,197 @@
+           f'mkdir -p {build.out} && {compile_step} || true'),
+      ],
+  }
+  build_lib.dockerify_run_step(compile_step,
+                               build,
+                               use_architecture_image_name=build.is_arm)
+  return compile_step
+
+
+def get_oss_fuzz_on_demand_build_steps(fuzzing_engine, project_image,
+                                       project_name):
+  steps = []
+  # TODO(metzman): Clone from the main branch.
+  steps.append(
+      build_lib.get_git_clone_step('https://github.com/google/fuzzbench.git',
+                                   'ood'))
+  builder_image_name = f'gcr.io/oss-fuzz-base/{fuzzing_engine}-builder',
+
+  build_args = [
+      'build', '--tag', f'gcr.io/oss-fuzz/{fuzzing_engine}/{project_name}',
+      '--build-arg', 'BUILDKIT_INLINE_CACHE=1', '--cache-from',
+      builder_image_name, '--build-arg', f'parent_image={project_image}',
+      '--file', f'fuzzbench/fuzzers/{fuzzing_engine}/builder.Dockerfile',
+      f'fuzzbench/fuzzers/{fuzzing_engine}'
+  ]
+  build_step = {
+      'args': build_args,
+      'name': build_lib.DOCKER_TOOL_IMAGE,
+  }
+  steps.append(build_step)
+  return steps
+
+
+def get_build_steps(  # pylint: disable=too-many-locals, too-many-arguments
+    project_name, project_yaml, dockerfile_lines, image_project,
+    base_images_project, config):
+  """Returns build steps for project."""
+  project = build_project.Project(project_name, project_yaml, dockerfile_lines,
+                                  image_project)
+  assert len(project.fuzzing_engines) == 1, project.fuzzing_engines
+  fuzzing_engine = project.fuzzing_engines[0]
+  if project.disabled:
+    logging.info('Project "%s" is disabled.', project.name)
+    return []
+
+  report_date = build_project.get_datetime_now().strftime('%Y%m%d')
+  bucket = CoverageBucket(project.name, report_date, PLATFORM, config.testing)
+
+  build_steps = build_lib.get_project_image_steps(project.name,
+                                                  project.image,
+                                                  project.fuzzing_language,
+                                                  config=config)
+  build_steps += get_oss_fuzz_on_demand_build_steps(fuzzing_engine,
+                                                    project.image, project.name)
+
+  build = build_project.Build(fuzzing_engine, 'address', ARCHITECTURE)
+  env = build_project.get_env(project.fuzzing_language, build)
+  logging.info('ood %s', config.oss_fuzz_on_demand)
+  build_steps.append(get_compile_step_ood(fuzzing_engine, project, build, env))
+  download_corpora_steps = build_lib.download_corpora_steps(
+      project.name, test_image_suffix=config.test_image_suffix)
+  if not download_corpora_steps:
+    logging.info('Skipping code coverage build for %s.', project.name)
+    return []
+
+  build_steps.extend(download_corpora_steps)
+
+  failure_msg = ('*' * 80 + '\nCode coverage report generation failed.\n'
```

Comment:
Where is this var used?

Reply:
I'm gonna remove this file for now. It's not ready for review.